internal/pkg/template/api/service/service.go

```diff
@@ -2,17 +2,55 @@ package service
 
 import (
 	"fmt"
+	"sync"
+	"time"
 
 	"github.com/keboola/keboola-as-code/internal/pkg/template/api/dependencies"
 	"github.com/keboola/keboola-as-code/internal/pkg/template/api/gen/templates"
+	"github.com/keboola/keboola-as-code/internal/pkg/template/repository/manager"
 )
 
 type Service struct {
-	dependencies dependencies.Container
+	dependencies        dependencies.Container
+	lock                *sync.Mutex
+	repositoriesManager *manager.Manager
 }
 
-func New(d dependencies.Container) templates.Service {
-	return &Service{dependencies: d}
+const TemplateRepositoriesPullInterval = 5 * time.Minute
+
+func New(d dependencies.Container) (templates.Service, error) {
+	repoManager, err := manager.New(d.Logger())
+	if err != nil {
+		return nil, err
+	}
+	s := &Service{
+		dependencies:        d,
+		lock:                &sync.Mutex{},
+		repositoriesManager: repoManager,
+	}
+	s.StartCron()
+	return s, nil
+}
+
+func (s *Service) StartCron() {
```

Comment:
@webrouse Kopíruju ten cron z webhooků, to je super, akorát teda dívám se že tam je to nějak jinak s tím cancel contextem: https://github.com/keboola/temp-webhooks-api/blob/74eb08f3a83259f1c516228b535d4d4757a8d27a/cmd/api/main.go#L46-L47 a tady je pro změnu `CtxCancelFn()`. Mám to nějak řešit, nebo je to takhle ok? 🙂

Reply:
Ahoj. Tvoj "konstruktor" by mal dosta iba `ctx context.Context` a dalej sa nestarat ... + mat tam ten `select` a `ctx.Done`.

To je uloha kodu nad tym spravit `cancel` a to tam uz mame, ked sa HTTP server vypina, tak sa vypne aj tento cron.

Reply:
FYI tu sa riesi context cancel:
https://github.com/keboola/keboola-as-code/blob/f3a6ed5d4128a1c6070dffc848c0a123dfaf872e/cmd/templates/main.go#L95-L96

Reply:
Nó, 🤔 context se dá vzít z `dependencies.Ctx()`, takže asi dobrý takhle. Ale myslím že už ten `for` se `select`em chápu. Spouští se pořád dokola v tom zadaným intervalu až do té doby než je `ctx.Done() == true` (což se stane ve chvíli zavolání toho `cancel`) a potom skončí. Jde teda jen o to, aby se ten cron ukončil a nepokračoval dál po ukončení celé aplikace.